contracts/apps/accounting/FinanceApp.sol

```diff
@@ -0,0 +1,432 @@
+    struct Transaction {
+        address token;
+        uint amount;
+        address externalAddress;
+        string reference;
+        uint timestamp;
+        uint accountingPeriodId;  // in which accounting period did this occur
+        TransactionType _type;
+    }
+
+    enum TransactionType {
+        Deposit,
+        Withdrawal
+    }
+
+    struct Payment {
+        ERC20 token;
+        uint amount;
+        address to;
+        uint repeat; // number of times this can be paid
+        uint timesCalled; //
+        uint startTimestamp; // Initial timestamp 
+        uint nextTimestamp; 
+        bool canceled; // override to cancel a payment
+        bytes2 ct_sec;
+        bytes2 ct_min;
+        bytes2 ct_hour;
+        bytes2 ct_day;
+        bytes2 ct_month;
+        bytes2 ct_weekday;
+        bytes2 ct_year;
+    }
+
+    Payment[] payments;
+
+
+
+    // The state a transaction update can be.
+    // New states should be added to the end to maintain the
+    // order of the index when interfacing with web3.
+    enum TransactionState {
+        New, // not needed?
+        PendingApproval,
+        Failed,
+        Succeeded,
+        Approved,
+        Denied
+    }
+
+    // The change in Transaciton state over time
+    struct TransactionUpdate {
+        uint transactionId; // Parent Transaction
+        TransactionState state;
+        string reason;
+        address actor; // who performed this update
+    }
+
+    Transaction[] public transactions;
+    TransactionUpdate[] public transactionUpdates;
+
+    // throttledFunctions[string key] = timstamp last run
+    mapping (string => uint) throttledFunctions;
+
+    // Reverse relation of a Transaction ID  -> TransactionsUpdatesIds[]
+    // transactionUpdatesRelation[tid] = [tuid_0..tuid_N]
+    mapping (uint => uint[]) public transactionUpdatesRelation;
+
+    event WithdrawalFailed(uint transactionId);
+
+    function () {
+    }
+
+    function FinanceApp() {
+    }
+
+    /**
+    @param vaultAddress The vault app to use with this FinanceApp
+    */
+    function initialize(address vaultAddress) onlyInit {
+        initialized();
+        vault = Vault(vaultAddress);
+    }
+
+    event NewPayment(uint pid);
+    /**
+    @notice This will a create a new payment
+    @param token The token that will be paid
+    @param amount The amount to be paid 
+    @param repeat This is the number of times that the payment should be sent 1..n times
+    @param startTimestamp This is when the payments will begin
+    */
+    function newPayment(ERC20 token, uint amount, address to, uint repeat, uint startTimestamp, bytes2 ct_sec, bytes2 ct_min, bytes2 ct_hour, bytes2 ct_day, bytes2 ct_month, bytes2 ct_weekday, bytes2 ct_year) auth external {
+        require(repeat > 0);
+        uint pid = _newPayment(token, amount, to, startTimestamp);
+        _setPaymentSchedual(pid, repeat, ct_sec, ct_min, ct_hour, ct_day, ct_month, ct_weekday, ct_year);
```

Comment:
If a payment can be withdrawn on creation, we should make it happen so only one transaction is needed for individual payments.

Reply:
So you mean if `repeat=1` then do the `token.transfer()` immediately?

Reply:
Actually if a transaction can happen at the moment of payment creation, we should execute the transfer.